src/common/types/types.cpp

```diff
@@ -363,32 +363,34 @@ void LogicalType::setPhysicalType() {
 
 LogicalType LogicalTypeUtils::dataTypeFromString(const std::string& dataTypeString) {
```

Comment:
I think it's better to separate the actual parsing of fixed list, var list, struct, map types into different functions, as this function is becoming larger now.